### DISCUSSION TITLE
Keycloak Grant Acess page customization

### DISCUSSION THREAD

**Discussion by tiagogalvao7** [keycloakify/keycloakify#606] [General]
I'm developing a theme for Keycloak, and I've already been able to customize the login page, the page to create a new user, but I still haven't managed to customize the "grant access" page. Can you explain how I can do this? Supposedly the Login theme allows you to change the Grant access page. I already have the layout created, but I haven't been able to apply it to Keycloak yet. I'm talking about the page in the image below:

![image](https://github.com/user-attachments/assets/77b8a5a8-97b0-4ea6-86af-9b41f20341e6)
![image](https://github.com/user-attachments/assets/ceeac52f-36b7-45c4-94f0-d34813a67cae)

**Comment by garronej** [comment]
Hello @tiagogalvao7,  

Well first of, you shouldn't apply the theme on the master realm (keycloak) you should create your own realm. But I assume this is just for the screenshot.  

The page your are looking for is probably `login-login-oauth-grant.ftl`  

When unsure you can open the console and type `kcContext.pageId` it will tell you what page you're on:  

<img width="2041" alt="image" src="https://github.com/user-attachments/assets/4d13920c-a85d-496f-806e-8da9dcf5a718">
